### ISSUE TITLE
Export of readonly conanfile.py results in permission denied error

### ISSUE THREAD

**Issue by climblinne** [conan-io/conan#3638]
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

I am using Conan 1.8.0-dev, Win10 & Python3.6.

When having an readonly `conanfile.py` file (SVN locking mechanism make them readonly), we get the follwing error message:
`ERROR: [Errno 13] Permission denied: 'C:\\...\\user\\svn25\\export\\conanfile.py'`

Because `_export_conanfile` function uses `shutil.copy2` all stat info including readonly are taken. May be add an option, to ignore readonly flag.

**Comment by climblinne** [comment]
The error was detected, by playing around with SVN feature (#3192). But it could also happen with every read-only file.

**Comment by jgsogo** [comment]
We need to check this, if this happens for any read-only file, it is more like a bug than an issue related to an experimental feature (SCM for SVN).

**Comment by danimtb** [comment]
PR merged! It will be fixed in next 1.9 release. Thanks!

**Comment by jgsogo** [comment]
I'm reopening it as there is an ongoing discussion here: https://github.com/conan-io/conan/pull/3748#issuecomment-432730079. I don't like to _close_ v1.9 without noticing what is the decision taken there.

**Comment by memsharded** [comment]
This has been reverted, because it is not possible to change permissions in that way.
It is not clear why is happening with SVN and not with other RO permissions. We need to further investigate it, but we are opening 1.9 branch today, so moving this to investigate for 1.10.

**Comment by jgsogo** [comment]
I'm trying to reproduce it in Mac and I'm not able to. It seems to happen only in Windows.

Having a locked file:

```
⇒  svn lock conanfile.py
svn: warning: W160035: Path '/conanfile.py' is already locked by user 'jgsogo' in filesystem '5d1b4008-cece-4d52-b44a-c9b3911d192e'
svn: E200009: One or more locks could not be obtained
```

and exporting the recipe twice, it works:

```
⇒  conan export . issue/test                                  
DEBUG :export.py      [45]: Exporting /Users/jgsogo/dev/conan/issues/3748/repo-wc/repo/conanfile.py [2018-10-29 11:50:52,479]
Exporting package recipe
name/ver@issue/test: WARN: Repo origin looks like a local path: file:///Users/jgsogo/dev/conan/issues/3748/repo@2
name/ver@issue/test: Repo origin deduced by 'auto': file:///Users/jgsogo/dev/conan/issues/3748/repo@2
name/ver@issue/test: Revision deduced by 'auto': 2
name/ver@issue/test: The stored package has not changed


⇒  conan export . issue/test
DEBUG :export.py      [45]: Exporting /Users/jgsogo/dev/conan/issues/3748/repo-wc/repo/conanfile.py [2018-10-29 11:50:56,463]
Exporting package recipe
name/ver@issue/test: WARN: Repo origin looks like a local path: file:///Users/jgsogo/dev/conan/issues/3748/repo@2
name/ver@issue/test: Repo origin deduced by 'auto': file:///Users/jgsogo/dev/conan/issues/3748/repo@2
name/ver@issue/test: Revision deduced by 'auto': 2
name/ver@issue/test: The stored package has not changed
```

**Comment by jgsogo** [comment]
I've created several tests that do not fail (#3868).

In your example, you was locking every file with ``svn propset svn:needs-lock * . -R`` in the _remote_ itself. Can the issue be associated to it? Are you able to replicate it locking in a working copy (which I think is the normal usage)?

**Comment by climblinne** [comment]
Yes, we always have several repositories, where all files use the svn locking system. 
The `svn:needs-lock` will first work, when you commit the property into the repository (then the file became 'read-only').

**Comment by jgsogo** [comment]
(I've added more tests, see the PR --> ) All tests pass except the ones where the _root_ folder is locked 😕

**Comment by jgsogo** [comment]
Please, check #3868: the only tests that fail are those that are adding the `needs-lock` property to the root of the working copy AND to the `conanfile.py` itself. Adding it to subfolders or files is ok.

Given the meaning of `needs-lock` property: it just tells SVN that the file ought to be locked before editing, but It does NOT lock the file, the file is readonly until executed `svn lock` over it; it looks like we do not need to change permissions, we can just remove this property in the working copy of the cache.

**Comment by jgsogo** [comment]
...so, the only thing needed is to make writable the `conanfile.py` in order to override the `auto` keywords and only if `needs-lock` property is set. As this conanfile is in the _export folder_ (it is not the one in the repository), we can only change its permissions.

**Comment by jgsogo** [comment]
It looks like it is not exactly the same issue. For SVN the error raises for file `...export\\conanfile.py` (when substituting the scm `auto` keyword), while in your log it seems to be the `...source/conanfile.py`, when copying/cloning your repo there. 

Are you using SCM?

**Comment by ehargitt** [comment]
Yes, we use Perforce for SCM (is that what you mean?). Sorry if this is in the wrong place. I only saw same file, same error, and caused also caused by read-only permission.

**Comment by jgsogo** [comment]
Don't worry, it's ok. I'm just trying to know about your issue. As you said, it should be behaving exactly the same because Perforce makes files read only too, but it is failing for you at a different point in the Conan workflow.

When I asked if you are using Perforce for SCM I mean if you are using the [scm attribute](https://docs.conan.io/en/latest/reference/conanfile/attributes.html#scm-attribute), I didn't explain myself, sorry. In order to use Conan with Perforce for this feature, you should have modified some lines in Conan sources and created the Python wrapper around Perforce command line (we don't provide it ourselves because noone of us uses Perforce and nobody has asked for it, maybe it is the time to do it).

Said that, I'm thinking that we haven't done any test with SVN as CVS (neither with Perforce) without using it with our scm-feature too 🤔... and that could be exactly your use case. I'll try to reproduce it with SVN, if not, it could be the excuse to try Perforce (I've never used it).
